test-count.ps1

```diff
@@ -0,0 +1,39 @@
+$e2ePath = "Tests/E2E"
+$integrationPath = "Tests/Integration" 
+$unitPath = "Tests/Unit"
+
+$testAttributeRegex = "\[ *(Fact|Theory) *\]"
+
+$outputFile = "test-count.csv"
+$fullOutputPath = Join-Path -Path $PWD -ChildPath $outputFile
+$gitRepoPath = "C:\sources\lora\iotedge-lorawan"
```

Comment:
You can easily convert this into a _cmdlet_ by promoting the variables into defaulted parameters:

```suggestion
[CmdletBinding()]
param (
  [string]$GitRepoPath = '.',
  [string]$E2ePath = 'Tests/E2E',
  [string]$IntegrationPath = 'Tests/Integration' ,
  [string]$UnitPath = 'Tests/Unit',
  [string]$TestAttributeRegex = '\[ *(Fact|Theory) *\]',
  [string]$OutputFile = 'test-count.csv'
)

$ErrorActionPreference = 'Stop'

$fullOutputPath = Join-Path -Path $PWD -ChildPath $outputFile
```

An immediate benefit is that you get automatic help and CLI arguments parsing support:

    PS> help .\test-count.ps1
    test-count.ps1 [[-GitRepoPath] <string>] [[-E2ePath] <string>] [[-IntegrationPath] <string>] [[-UnitPath] <string>] [[-TestAttributeRegex] <string>] [[-OutputFile] <string>] [<CommonParameters>]


I have also suggested to set `$ErrorActionPreference` to stop on any error.